packages/core/src/interpreter.ts

```diff
@@ -400,6 +400,9 @@ export class Interpreter<
 
     this._state = resolvedState;
 
+    // TODO: this notifies all subscribers but usually this is redundant
+    // if we are using the initialState as `resolvedState` then there is no real change happening here
+    // we need to rethink if this needs to be refactored
```

Comment:
So is it only useful for when the `resolvedState` is different from the initial state?

Makes me wonder if `initialState` is being passed in too late, because it doesn't make sense for an observer to get:

1. The initial state of the machine
2. Immediately after, an arbitrary restored state

Ideally, it should never get 1. So maybe we should think about revising the API:

```js
interpret(machine, { state: restoredState })
  .start(); // never pass in restored state here
```